src/main/java/org/usfirst/frc/team2473/framework/Devices.java

```diff
@@ -293,23 +296,26 @@ public void removeDoubleSolenoid(int forward, int reverse) {
 		if(doubleSolenoids.containsKey(forward + " " + reverse)) doubleSolenoids.remove(forward + " " + reverse);				
     }
     
-    public void initializeCVSocket() {
-        if (cvSocket == null) {
+    public void updateVisionValues() {
```

Comment:
Given that you need to initialize the serial connection, check it regularly, and keep state information, it might be a bit much to lump in with the rest of the global Device object. If this was C++ maybe I'd let it slide, but since we're working with Java let's make a dedicated class for Serial stuff.